modules/unsupported/csv/src/main/java/org/geotools/data/csv/CSVFeatureWriter.java

```diff
@@ -0,0 +1,174 @@
+    }
+    
+    // featureType start
+    @Override
+    public SimpleFeatureType getFeatureType() {
+        return this.featureType;
+    }
+    // featureType end
+
+    // hasNext start
+    @Override
+    public boolean hasNext() throws IOException {
+        if( csvWriter == null ){
+            throw new IOException("Writer has been closed");
+        }
+        if (this.appending) {
+            return false; // reader has no more contents
+        }
+        return iterator.hasNext();
+    }
+    // hasNext end
+    
+    // next start
+    @Override
+    public SimpleFeature next() throws IOException, IllegalArgumentException,
+            NoSuchElementException {
+        if(csvWriter == null){
+            throw new IOException("Writer has been closed");
+        }
+        if (this.currentFeature != null) {
+            this.write(); // the previous one was not written, so do it now.
+        }
+        try {
+            if(!appending){
+                if(iterator.hasNext()){
+                    this.currentFeature = iterator.next();
+                    return this.currentFeature;
+                }
+                else {
+                    this.appending = true;
+                }
+            }
+            String fid = featureType.getTypeName()+"."+nextRow;
+            Object values[] = DataUtilities.defaultValues( featureType );
+            
+            this.currentFeature = SimpleFeatureBuilder.build( featureType, values, fid );
+            return this.currentFeature;
+        }
+        catch (IllegalArgumentException invalid ){
+            throw new IOException("Unable to create feature:"+invalid.getMessage(),invalid);
+        }
+    }
+    // next end
+    
+    // remove start
+    /**
+     * Mark our {@link #currentFeature} feature as null, it will be skipped when written effectively removing it.
+     */
+    public void remove() throws IOException {
+        this.currentFeature = null; // just mark it done which means it will not get written out.
+    }
+    // remove end
+    
+    // write start
+    public void write() throws IOException {
+        if (this.currentFeature == null) {
+            return; // current feature has been deleted
+        }
+        this.csvWriter.writeRecord(this.csvStrategy.encode(this.currentFeature));
+        nextRow++;
+        this.currentFeature = null; // indicate that it has been written
+    }
+    // write end
+    
+    // close start
+    @Override
+    public void close() throws IOException {
+        if( csvWriter == null ){
+            throw new IOException("Writer alread closed");
+        }
+        if (this.currentFeature != null) {
+            this.write(); // the previous one was not written, so do it now.
+        }
+        // Step 1: Write out remaining contents (if applicable)
+        while (hasNext()) {
+            next();
+            write();
+        }
+        csvWriter.close();
```

Comment:
The [docs for CsvWriter](http://javacsv.sourceforge.net/com/csvreader/CsvWriter.html#close()) says it does, and I did a quick test, and confirmed. It does close whatever you pass it. :)